packages/graphiql/src/components/QueryEditor.tsx

```diff
@@ -83,210 +108,172 @@ export class QueryEditor extends React.Component<QueryEditorProps, {}>
-          this.props.onClickReference && this.props.onClickReference(reference),
+          props.onClickReference && props.onClickReference(reference),
       },
       jump: {
-        schema: this.props.schema,
+        schema,
         onClick: (
           reference: GraphQLType, // TODO: it looks like this arg is not actually a GraphQL type but something from GraphiQL codemirror
-        ) =>
-          this.props.onClickReference && this.props.onClickReference(reference),
+        ) => props.onClickReference && props.onClickReference(reference),
       },
       gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
       extraKeys: {
         'Cmd-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+            container: editorEl,
+          }),
         'Ctrl-Space': () =>
-          // @ts-ignore showHint method needs improvement on definatelytyped
-
-          editor.showHint({ completeSingle: true, container: this._node }),
+          // showHint method needs improvement on definatelytyped
+          // @ts-ignore
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+            container: editorEl,
+          }),
         'Alt-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+
+            container: editorEl,
+          }),
         'Shift-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
+
+            container: editorEl,
+          }),
         'Shift-Alt-Space': () =>
           // @ts-ignore showHint method needs improvement on definatelytyped
-          editor.showHint({ completeSingle: true, container: this._node }),
+          editorRef.current.showHint({
+            completeSingle: true,
+            // @ts-ignore
 
-        'Cmd-Enter': () => {
-          if (this.props.onRunQuery) {
-            this.props.onRunQuery();
-          }
-        },
+            container: editorEl,
+          }),
+
+        'Cmd-Enter': () => session.executeOperation(),
         'Ctrl-Enter': () => {
-          if (this.props.onRunQuery) {
-            this.props.onRunQuery();
-          }
+          console.log(session.operation.text);
+          session.executeOperation();
         },
 
         'Shift-Ctrl-C': () => {
-          if (this.props.onCopyQuery) {
-            this.props.onCopyQuery();
+          if (props.onCopyQuery) {
```

Comment:
I could be wrong, but I believe that there is a danger here of capturing a stale version of props

Reply:
If you want to use the most up to date version of props, while still only calling use effect once, you may wish to create:
`propsRef = useRef(props); useEffect(() => { propsRef.current = props; }, [props]);`

Reply:
this is done on `QueryEditor.didMount()`, so we only initialize the editor once on mount.